prov/rxm/src/rxm_cq.c

```diff
@@ -61,30 +61,30 @@ static const char *rxm_cq_strerror(struct fid_cq *cq_fid, int prov_errno,
 	return fi_cq_strerror(rxm_ep->msg_cq, prov_errno, err_data, buf, len);
 }
 
-/* Get a match_iov derived from iov whose size matches given length */
 static int rxm_match_iov(const struct iovec *iov, void **desc,
 			 uint8_t count, uint64_t offset, size_t match_len,
 			 struct rxm_iov *match_iov)
 {
-	uint8_t i;
+	uint8_t i, j;
 
 	assert(count <= RXM_IOV_LIMIT);
 
-	for (i = 0; i < count; i++) {
+	for (i = 0, j = 0; i < count; i++) {
 		if (offset >= iov[i].iov_len) {
 			offset -= iov[i].iov_len;
 			continue;
 		}
 
-		match_iov->iov[i].iov_base = (char *)iov[i].iov_base + offset;
-		match_iov->iov[i].iov_len = MIN(iov[i].iov_len - offset, match_len);
+		match_iov->iov[j].iov_base = (char *)iov[i].iov_base + offset;
+		match_iov->iov[j].iov_len = MIN(iov[i].iov_len - offset, match_len);
 		if (desc)
-			match_iov->desc[i] = desc[i];
+			match_iov->desc[j] = desc[i];
 
-		match_len -= match_iov->iov[i].iov_len;
+		match_len -= match_iov->iov[j].iov_len;
 		if (!match_len)
 			break;
```

Comment:
This is existing code, but we can jump past the if (match_len) statement that follows the for loop to avoid a branch that will never be taken.